services/users.js

```diff
@@ -96,12 +97,20 @@ async function deleteUserAndFriends (secret) {
 }
 
 async function removeFriend (userId, friendId, secret) {
-  const authentication = await authenticateUser(userId, secret)
+  const authentication = await UserModel.authenticateUser(userId, secret)
   if (authentication) {
     return UserModel.removeFriendship(userId, friendId)
   }
 }
 
+async function getUserIdsFromSecrets (secrets) {
+  const userIds = {}
+  for (let i = 0; i < secrets.length; i++) {
+    userIds[secrets[i]] = await UserModel.getUserIdBySecret(secrets[i])
+  }
+  return userIds
+}
```

Comment:
smart

Reply:
To be even smarter, you can get the secrets in parallel by building and awaiting a single `Promise.all`, rather than awaiting the first secret, and only starting to get the second secret once the first one resolves.

To be even smarter, you can try to bulk query sql. Maybe look into `sqlize` (postgres ORM, which basically allows us to call fancy functions which return promises instead of writing sql ourselves)